internal/cli/compile/compile.go

```diff
@@ -293,48 +270,57 @@ func runCompileCommand(cmd *cobra.Command, args []string) {
 			libs += fmt.Sprintln("      - " + lib.GetName() + " (" + lib.GetVersion() + ")")
 		}
 		if hasVendoredLibs {
-			fmt.Println()
-			fmt.Println(tr("WARNING: The sketch is compiled using one or more custom libraries."))
-			fmt.Println(tr("Currently, Build Profiles only support libraries available through Arduino Library Manager."))
+			msg := "\n"
+			msg += tr("WARNING: The sketch is compiled using one or more custom libraries.") + "\n"
+			msg += tr("Currently, Build Profiles only support libraries available through Arduino Library Manager.")
+			feedback.Warning(msg)
 		}
 
 		newProfileName := "my_profile_name"
 		if split := strings.Split(compileRequest.GetFqbn(), ":"); len(split) > 2 {
 			newProfileName = split[2]
 		}
-		fmt.Println()
-		fmt.Println("profiles:")
-		fmt.Println("  " + newProfileName + ":")
-		fmt.Println("    fqbn: " + compileRequest.GetFqbn())
-		fmt.Println("    platforms:")
+		profile := fmt.Sprintln()
+		profile += fmt.Sprintln("profiles:")
+		profile += fmt.Sprintln("  " + newProfileName + ":")
+		profile += fmt.Sprintln("    fqbn: " + compileRequest.GetFqbn())
+		profile += fmt.Sprintln("    platforms:")
 		boardPlatform := compileRes.GetBoardPlatform()
-		fmt.Println("      - platform: " + boardPlatform.GetId() + " (" + boardPlatform.GetVersion() + ")")
+		profile += fmt.Sprintln("      - platform: " + boardPlatform.GetId() + " (" + boardPlatform.GetVersion() + ")")
 		if url := boardPlatform.GetPackageUrl(); url != "" {
-			fmt.Println("        platform_index_url: " + url)
+			profile += fmt.Sprintln("        platform_index_url: " + url)
 		}
 
 		if buildPlatform := compileRes.GetBuildPlatform(); buildPlatform != nil &&
 			buildPlatform.Id != boardPlatform.Id &&
 			buildPlatform.Version != boardPlatform.Version {
-			fmt.Println("      - platform: " + buildPlatform.GetId() + " (" + buildPlatform.GetVersion() + ")")
+			profile += fmt.Sprintln("      - platform: " + buildPlatform.GetId() + " (" + buildPlatform.GetVersion() + ")")
 			if url := buildPlatform.GetPackageUrl(); url != "" {
-				fmt.Println("        platform_index_url: " + url)
+				profile += fmt.Sprintln("        platform_index_url: " + url)
 			}
 		}
 		if len(libs) > 0 {
-			fmt.Println("    libraries:")
-			fmt.Print(libs)
+			profile += fmt.Sprintln("    libraries:")
+			profile += fmt.Sprint(libs)
 		}
+
+		// Output profile as a result
+		if _, err := stdOut.Write([]byte(profile)); err != nil {
+			feedback.FatalError(err, feedback.ErrGeneric)
```

Comment:
Is there any case where `stdOut.Write` could return an error under normal conditions (i.e., there isn't a bug in the Arduino CLI codebase)?

If not, it seems a panic would be more appropriate here so that the user can provide the stack trace to assist the investigation.

Reply:
It depends on what's behind the `io.Writer`, but in general yes (for example, a disk full when writing on a file, just to tell the first thing that comes to mind).

So we cannot panic, we must report the error gracefully.